include/cppkafka/utils/buffered_producer.h

```diff
@@ -772,6 +819,13 @@ void BufferedProducer<BufferType, Allocator>::produce_message(BuilderType&& buil
             if (ex.get_error() == RD_KAFKA_RESP_ERR__QUEUE_FULL) {
                 // If the output queue is full, then just poll
                 producer_.poll();
+                // Notify application so it can slow-down production
+                if (queue_full_notify) {
+                    queue_full_notify = queue_full_trigger_ = false; //clear trigger and local state
```

Comment:
This is always resetting `queue_full_notify`, so if you actually use `QueueFullNotification::EachOccurence` you'll still get only once per `produce_message` call, right? Not sure if the issue is here or if you wanted "each occurrence" to still mean "every time but only once per message"/